### PR TITLE
Consumer scheduling scaling KafkaChannel tests

### PR DESCRIPTION
**Pull request by aavarghese**
Signed-off-by: aavarghese <avarghese@us.ibm.com>

Part of #1537 
Dependent on #1618's implementation

## Proposed Changes

- Adding new KafkaChannel reconciler tests

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by aavarghese** [comment]
ping @pierDipi :)

**Comment by aavarghese** [comment]
Rebased and updated as per review comment changes in Channel reconciler (#1618)

File: control-plane/pkg/reconciler/channel/v2/channelv2_test.go [hunk 1]
```diff
@@ -0,0 +1,1237 @@
...
+			},
+			WantEvents: []string{
+				finalizerUpdatedEvent,
+				Eventf(
+					corev1.EventTypeWarning,
+					"InternalError",
+					"Subscription not ready. Failed to reconcile consumer group. Consumer group not ready",
+				),
+			},
+		},
```
[the remark this answers is not in the document]
Comment (aavarghese) [reply control-plane/pkg/reconciler/channel/v2/channelv2_test.go]: @pierDipi I agree. I have a more relaxed condition report in the latest commit. Can you TAL? This is also now more consistent with the implementation in KafkaSource and Trigger, as well. Thanks for the suggestion.

**Comment by aavarghese** [comment]
ping @pierDipi :)

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

File: control-plane/pkg/reconciler/channel/testdata/testdata [hunk 2]
```diff
@@ -0,0 +1 @@
+testdata
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/channel/testdata/testdata:1]: We broke some GH actions tests with this symlink :(

Comment (aavarghese) [reply control-plane/pkg/reconciler/channel/testdata/testdata:1]: Looking...

**Pull request merged**